agent/engine/dockerclient/dockerclientfactory.go

```diff
@@ -121,8 +137,66 @@ func findDockerVersions(endpoint string) map[DockerVersion]dockeriface.Client {
 		err = client.Ping()
 		if err != nil {
 			log.Infof("Failed to ping with Docker version %s: %v", version, err)
+			continue
 		}
 		clients[version] = client
 	}
 	return clients
 }
+
+func findDockerVersionsfromMinMaxVersions(endpoint string) (map[DockerVersion]dockeriface.Client, error) {
+	// get a Docker client with the default supported version
+	client, err := newVersionedClient(endpoint, string(minDockerAPILinux))
+	if err != nil {
+		return nil, errors.Wrap(err, "Error while creating client")
```

Comment:
Please add more context to the error and not use upper case characters in the message. 
The code base has plenty of examples using `errors.Wrap`